File: app/src/main/java/com/example/myplaylist/MainActivity.java
package com.example.myplaylist;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.MenuItem;
import android.widget.LinearLayout;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.bottomnavigation.BottomNavigationView;
public class MainActivity extends AppCompatActivity {
    final String TAG = this.getClass().getSimpleName();

    LinearLayout home_ly;
    BottomNavigationView bottomNavigationView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        init(); //객체 정의
        SettingListener(); //리스너 등록

        //맨 처음 시작할 탭 설정
        bottomNavigationView.setSelectedItemId(R.id.item_fragment1);
    }

    private void init() {
        home_ly = findViewById(R.id.home_ly);
        bottomNavigationView = findViewById(R.id.bottomNavigationView);
    }

    private void SettingListener() {
        //선택 리스너 등록
        bottomNavigationView.setOnItemSelectedListener(new TabSelectedListener());
    }

    class TabSelectedListener implements BottomNavigationView.OnNavigationItemSelectedListener{
        @SuppressLint("NonConstantResourceId")
        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
            switch (menuItem.getItemId()) {
                case R.id.item_fragment1: {
                    getSupportFragmentManager().beginTransaction()
                            .replace(R.id.home_ly, new FragmentPage1())
                            .commit();
                    return true;
                }
                case R.id.item_fragment2: {
                    getSupportFragmentManager().beginTransaction()
                            .replace(R.id.home_ly, new FragmentPage2())
                            .commit();
                    return true;
                }
                case R.id.item_fragment3: {
                    getSupportFragmentManager().beginTransaction()
                            .replace(R.id.home_ly, new FragmentPage3())
                            .commit();
                    return true;
                }
            }

            return false;
        }
    }
}

File: app/src/main/java/com/example/myplaylist/MainActivity.kt
package com.example.myplaylist

import android.view.LayoutInflater
import android.os.Bundle
import com.example.myplaylist.R
import androidx.appcompat.app.AppCompatActivity
import android.widget.EditText
import android.content.Intent
import com.example.myplaylist.RegisterPage
import android.widget.Toast
import com.example.myplaylist.MainActivity
import org.json.JSONException
import com.example.myplaylist.LoginRequest
import com.android.volley.RequestQueue
import com.android.volley.toolbox.Volley
import com.android.volley.toolbox.StringRequest
import com.android.volley.AuthFailureError
import android.widget.LinearLayout
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.example.myplaylist.MainActivity.TabSelectedListener
import android.annotation.SuppressLint
import android.view.MenuItem
import com.example.myplaylist.FragmentPage1
import com.example.myplaylist.FragmentPage2
import com.example.myplaylist.FragmentPage3

class MainActivity : AppCompatActivity() {




    val TAG = this.javaClass.simpleName
    private var home_ly: LinearLayout? = null
    private var bottomNavigationView: BottomNavigationView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        init() //객체 정의
        SettingListener() //리스너 등록

        //맨 처음 시작할 탭 설정
        bottomNavigationView!!.selectedItemId = R.id.item_fragment1
    }

    private fun init() {
        home_ly = findViewById(R.id.home_ly)
        bottomNavigationView = findViewById(R.id.bottomNavigationView)
    }

    private fun SettingListener() {
        //선택 리스너 등록
        bottomNavigationView!!.setOnItemSelectedListener(TabSelectedListener())
    }

    internal inner class TabSelectedListener :
        BottomNavigationView.OnNavigationItemSelectedListener {
        @SuppressLint("NonConstantResourceId")
        override fun onNavigationItemSelected(menuItem: MenuItem): Boolean {
            when (menuItem.itemId) {
                R.id.item_fragment1 -> {
                    supportFragmentManager.beginTransaction()
                        .replace(R.id.home_ly, FragmentPage1())
                        .commit()
                    return true
                }
                R.id.item_fragment2 -> {
                    supportFragmentManager.beginTransaction()
                        .replace(R.id.home_ly, FragmentPage2())
                        .commit()
                    return true
                }
                R.id.item_fragment3 -> {
                    supportFragmentManager.beginTransaction()
                        .replace(R.id.home_ly, FragmentPage3())
                        .commit()
                    return true
                }
            }
            return false
        }
    }
}